File: template/common/src/main/java/com/example/modid/ModInit.java
package com.example.modid;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

// shared, server-safe logic starts here
public final class ModInit {
  private static final String MOD_ID = "modid";
  private static final Logger LOGGER = LoggerFactory.getLogger(MOD_ID);

  private ModInit() {}

  public static void init() {
    LOGGER.info("{} common init", MOD_ID);
  }
}

File: template/common/src/main/kotlin/com/example/modid/ModInit.kt
package com.example.modid

import org.slf4j.LoggerFactory

// shared, server-safe logic starts here
object ModInit {
  private const val MOD_ID = "modid"
  private val logger = LoggerFactory.getLogger(MOD_ID)

  @JvmStatic
  fun init() {
    logger.info("{} common init", MOD_ID)
  }
}
